program/src/state/obligation.rs

```diff
@@ -83,56 +90,139 @@ impl Obligation {
         Ok(())
     }
 
-    /// Calculate obligation health ratio
-    pub fn calc_health(&self) -> Result<u64, ProgramError> {
+    /// Calc pending interest amount
+    /// borrowed * (current_slot - interest_slot) * interest
+    pub fn calc_pending_interest_amount(
+        &self,
+        slot: Slot,
+        interest: u64,
+    ) -> Result<u64, ProgramError> {
+        let slot_offset = slot
+            .checked_sub(self.interest_slot)
+            .ok_or(LendingError::CalculationFailure)?;
+
+        let pending = (self.amount_liquidity_borrowed as u128)
+            .checked_mul(slot_offset as u128)
+            .ok_or(LendingError::CalculationFailure)?
+            .checked_mul(interest as u128)
+            .ok_or(LendingError::CalculationFailure)?
+            .checked_div(INTEREST_POWER as u128)
+            .ok_or(LendingError::CalculationFailure)? as u64;
+
+        Ok(pending)
+    }
+
+    /// Calc effective interest amount
+    /// interest_amount + borrowed * (current_slot - interest_slot) * interest
+    pub fn calc_effective_interest_amount(
+        &self,
+        slot: Slot,
+        interest: u64,
+    ) -> Result<u64, ProgramError> {
+        let amount = self
+            .interest_amount
+            .checked_add(self.calc_pending_interest_amount(slot, interest)?)
+            .ok_or(LendingError::CalculationFailure)?;
+
+        Ok(amount)
+    }
+
+    /// Update intereset per each borrow
+    pub fn update_interest_amount(&mut self, amount: u64) {
+        self.interest_amount = amount;
```

Comment:
Looked through the code and thought we can split it into two different functions: one for accruing interest and another for repaying one (as we do for borrow/repay for example)